### PR TITLE
Add OpenTelemetry tracing to Envoy example

### PR DESCRIPTION
**Pull request by smithclay**
Updates Envoy example to show how to use the (new) OpenTelemetry-based tracer in Envoy with a basic collector.

image

### CODE REVIEW

File: collector/envoy/docker-compose.override.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ services:
   loadgen:
     depends_on: [nginx_proxy]
     image: williamyeh/hey
-    command: ["-z", "10m", "-c", "50", "-q", "50", "http://front-envoy:8080"]
+    command: ["-z", "10m", "-c", "50", "-q", "50", "http://localhost:8080/service/1"]
```
Comment (nslaughter) [inline collector/envoy/docker-compose.override.yaml:18]: This was useful at the time for moving the metrics on envoy, but I don't think we have to keep this loadgen shim.

**Review by nslaughter — COMMENTED** [review]
The way this example works I had to do `docker compose build flask_service` before I could run with `docker compose up`. Are the docs clear enough on that point to get this working.

File: collector/envoy/app/tracing/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,7 @@
 FROM flask_service:python-3.10-slim-bullseye
 
-COPY --from=envoyproxy/envoy-dev:latest /usr/local/bin/envoy /usr/local/bin/envoy
+# envoy v1.23.0
+COPY --from=envoyproxy/envoy-dev:4e2c7adfb393a8ea2a38c94494ccd2ee0fd469f9 /usr/local/bin/envoy /usr/local/bin/envoy
```
Comment (nslaughter) [inline collector/envoy/app/tracing/Dockerfile:4]: Do we benefit from pinning in these? I was thinking the drift will help monitor the status of integrations, but there are other ways.

**Review by nslaughter — APPROVED** [review]
Left a couple comments for consideration. Nothing blocking.

File: collector/envoy/app/tracing/Dockerfile (hunk 2 above)
Comment (smithclay) [reply collector/envoy/app/tracing/Dockerfile]: Agreed, removed :+1:

**Pull request merged**